iceoryx_dds/include/iceoryx_dds/dds/data_writer.hpp

```diff
@@ -27,44 +27,36 @@ namespace dds
 {
 ///
 /// @brief Abstraction for DDS Data Writers.
-/// Provides the minimum functionality required for posh-dds gateway implementations.
+/// @note Provides the minimum functionality required for posh-dds gateway implementations.
 ///
 class DataWriter
 {
   public:
+    /// @brief Virtual default destructor
     virtual ~DataWriter() = default;
 
-    ///
-    /// @brief connect Connect the DataWriter to the underlying DDS network.
-    ///
+    /// @brief Virtual Function to connect the DataWriter to the underlying DDS network.
```

Comment:
C++ has no virtual functions only virtual methods.
To provide you some context: `DataWriter` is an abstract base class that enforces a certain interface like for instance that every class which inherits from `DataWriter` has to implement a `connect` method.

I would get rid of the `Virtual` keyword at all since this is something the user sees directly in the signature of the method.